app/src/pages/playground/PlaygroundOutput.tsx

```diff
@@ -15,25 +20,61 @@ import {
   PlaygroundOutputSubscription$data,
   PlaygroundOutputSubscription$variables,
 } from "./__generated__/PlaygroundOutputSubscription.graphql";
+import { isChatMessages } from "./playgroundUtils";
 import { PlaygroundInstanceProps } from "./types";
 
 interface PlaygroundOutputProps extends PlaygroundInstanceProps {}
 
+export function PlaygroundOutputMessage({ message }: { message: ChatMessage }) {
+  const styles = useChatMessageStyles(message.role);
+
+  return (
+    <Card title={message.role} {...styles} variant="compact">
+      {message.content}
+    </Card>
+  );
+}
+
 export function PlaygroundOutput(props: PlaygroundOutputProps) {
   const instanceId = props.playgroundInstanceId;
-  const instance = usePlaygroundContext((state) => state.instances[instanceId]);
+  const instances = usePlaygroundContext((state) => state.instances);
+  const instance = instances.find((instance) => instance.id === instanceId);
   if (!instance) {
-    return null;
+    throw new Error("Playground instance not found");
   }
   const runId = instance.activeRunId;
   const hasRunId = runId !== null;
+
+  const OutputEl = useMemo(() => {
+    if (hasRunId) {
+      return (
+        <PlaygroundOutputText key={runId} playgroundInstanceId={instanceId} />
+      );
+    }
+    if (isChatMessages(instance.output)) {
+      const messages = instance.output;
+
+      return messages.map((message, index) => {
+        return <PlaygroundOutputMessage key={index} message={message} />;
+      });
+    }
```

Comment:
Oh as in this was the original output?

Reply:
yes, displaying the original output but then also displaying new outputs in the same way